src/app.js

```diff
@@ -276,34 +281,81 @@ fluid.onUncaughtException.addListener(function (err) {
  */
 
 fluid.defaults("gpii.app.tray", {
-    gradeNames: "fluid.component",
+    gradeNames: "fluid.modelComponent",
     members: {
         tray: {
             expander: {
                 funcName: "gpii.app.makeTray",
-                args: ["{that}.options.icon"]
+                args: ["{that}.model.icon"]
             }
+        },
+        icons: {
+            keyedIn: "icons/gpii-color.ico",
+            keyedOut: "icons/gpii.ico"
         }
     },
     components: {
         menu: {
             type: "gpii.app.menuInApp"
         }
     },
+    model: {
+        keyedInUserToken: null,
+        icon: "{that}.icons.keyedOut"
+    },
+    modelRelay: {
+        keyedInUserToken: {
+            target: "icon",
+            singleTransform: {
+                type: "fluid.transforms.free",
```

Comment:
I think we could use a conditional transform here and get rid of the 'getTrayIcon' function. 
http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#conditional-transform-fluid-transforms-condition-

Reply:
Yeah, it'd definitely make things prettier